pkg/testing/tools/artifacts_api.go

```diff
@@ -138,20 +136,17 @@ func WithHttpClient(client httpDoer) ArtifactAPIClientOpt {
 // More information about the API can be found at https://artifacts-api.elastic.co/v1
 // which will print a list of available operations
 type ArtifactAPIClient struct {
-	c      httpDoer
-	url    string
-	cdnURL string
-
-	logger logger
+	c       httpDoer
+	url     string
+	logFunc logFunc
```

Comment:
I understand that this is needed because of the logger that has been added to the ArtifactAPIClient but I was hoping for a cleaner solution 😅

Reply:
@pchila any better ideas? We use this client with 2 different logger interfaces: `t.Logf` in tests and `log.Printf` in the magefile. So, I came up with a common solution.

Reply:
I made this change to make this PR work (conflict resolution), refactoring of the artifact client is not the goal of this PR. We can follow up on this while working on https://github.com/elastic/elastic-agent/issues/4354